packages/grant-explorer/src/features/discovery/FilterDropdown.tsx

```diff
@@ -0,0 +1,192 @@
+import { Fragment } from "react";
+import { Link, useSearchParams } from "react-router-dom";
+import { Transition, Listbox } from "@headlessui/react";
+import {
+  ChevronDownIcon,
+  CheckIcon,
+  ChevronUpIcon,
+} from "@heroicons/react/20/solid";
+
+import { CHAINS } from "../api/utils";
+import { Dropdown, DropdownItem } from "../common/Dropdown";
+import { toURL } from "./RoundsFilter";
+
+type Option = {
+  label: string;
+  value: string;
+  children: Option[];
+};
+const filterOptions = [
+  {
+    label: "All",
+    value: "",
+    children: [],
+  },
+  {
+    label: "Type",
+    value: "type",
+    children: [
+      {
+        label: "Quadratic funding",
+        value: "MERKLE",
+      },
+      {
+        label: "Direct grants",
+        value: "DIRECT",
+      },
+    ],
+  },
+  {
+    label: "Status",
+    value: "status",
+    children: [
+      {
+        label: "Active",
+        value: "",
+      },
+      {
+        label: "Taking applications",
+        value: "apply",
+      },
+      {
+        label: "Finished",
+        value: "finished",
+      },
+    ],
+  },
+  {
+    label: "Network",
+    value: "network",
+    children: Object.entries(CHAINS).map(([value, { name }]) => ({
+      label: `Rounds on ${name}`,
+      value,
+    })),
+  },
+] as Option[];
+
+export type FilterProps = {
+  type: string;
+  status: string;
+  network: string;
+};
+
+/* 
+Find the label to display from the current filter.
+- All - nothing selected
+- Multiple - more than 1 selected
+- Selected - 1 selected
+*/
+function getLabel(filter: FilterProps) {
+  return (
+    // Convert { key: val } to [[key, val]] and remove empty values
+    Object.entries(filter)
+      .filter(([, val]) => Boolean(val))
+      .reduce(
+        (acc, [key, val], _, arr) => {
+          // More than 1 filter is selected
+          if (arr.length > 1) return { label: "Multiple" } as Option;
+
+          // Find the selected option
+          const selected =
+            filterOptions
+              .find((opt) => opt.value === key)
+              ?.children.find((c) => c.value === val) || acc;
+
+          return selected;
+        },
+        filterOptions[0] // Initialize with label: All
+      )
+  );
+}
+export function FilterDropdown({ status, type, network }: FilterProps) {
+  // Get existing search params
+  const params = Object.fromEntries(useSearchParams()[0]);
```

Comment:
Never seen a hook be used like this, and I'm pretty sure it's not _supposed_ to be used like this. @bard @gravityblast @bhargavaparoksham thoughts?

Reply:
Technically it's not incorrect but there's a small chance it might throw tools off that are relying on convention (like the linter).

I'd ask @carlbarrdahl  though why turn a search params object (I'm guessing [this](https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams)) into a less specialized data type here?

(I'll note this is still in draft though so I might be commenting on wip.)

Reply:
Gotcha. In that case how about this to make it extra clear?


```ts
const updatedFilterParams = new URLSearchParams({ 
  ...Object.fromEntries(params), 
  [item.value]: child.value 
});

// ...
<DropdownItem to={`/rounds?${updatedFilterParams.toString()}`}
```

Reply:
> Gotcha. In that case how about this to make it extra clear?
> 
> ```tsx
> const updatedFilterParams = new URLSearchParams({ 
>   ...Object.fromEntries(params), 
>   [item.value]: child.value 
> });
> 
> // ...
> <DropdownItem to={`/rounds?${updatedFilterParams.toString()}`}
> ```

That's what the utility function does:
```ts
export const toURL = (filterParams: Partial<Filter> = {}) =>
  new URLSearchParams(filterParams).toString();
``` 

I guess we could change it to accept the URLSearchParams and convert it into and object and merge from there:
```ts
export const toURL = (
  filterParams: Partial<Filter> = {},
  previousParams: URLSearchParams
) =>
  new URLSearchParams({
    ...Object.fromEntries(previousParams),
    ...filterParams,
  }).toString();
...
toURL({ [item.value]: child.value }, params)
```
or event create a hook that contains the params and merges it with the new object:
```ts
export function useMergeQuery() {
  const [params] = useSearchParams();
  return (filter: Partial<Filter>) =>
    new URLSearchParams({
      ...Object.fromEntries(params),
      ...filter,
    }).toString();
}

const toURL = useMergeQuery()
...
```

Reply:
I wouldn't go with the last one (as Agent Smith would say, don't send a hook to do a function's job) but other than that, any implementation is fine; only thing is, I'd consider making the name truer to its function with something like "toUrlSearchString" or "toQueryString", since what it's producing is... well, not a URL. :)